src/pybind/mgr/cephadm/module.py

```diff
@@ -229,7 +231,7 @@ def trivial_result(val):
     return AsyncCompletion(value=val, name='trivial_result')
 
 
-class CephadmOrchestrator(MgrModule, orchestrator.Orchestrator):
+class CephadmOrchestrator(MgrModule, orchestrator.OrchestratorClientMixin):
```

Comment:
In addition to having `wait`, OrchestratorClientMixin also contains magic:

https://github.com/ceph/ceph/blob/0f0ae55a0d1dd92612dab81f1ed0cc0ded703bbe/src/pybind/mgr/orchestrator.py#L1414-L1426

It overwrites all methods defined in `Orchestrator` redirects them all to the orchestrator implementation. Which means that you'll get nice `RuntimeError: maximum recursion depth exceeded` (or somethimg similar) errors instead of `AttributeError`s for not implemented features.

My idea would be to remove `_orchestrator_wait` from OrchestratorClientMixin

Reply:
Rethinking this again. Yes please. `OrchestratorClientMixin` is meant for the dashboard and for the cli. no one else should use it by inheriting from it. It's ok to use it, but please don't inherent from it.